DrawerKit/DrawerKit/Public API/Notifications.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+public class NotificationToken {
+    public let token: NSObjectProtocol
+    public let center: NotificationCenter
+
+    init(token: NSObjectProtocol, center: NotificationCenter) {
+        self.token = token
+        self.center = center
+    }
+    
+    deinit { center.removeObserver(token) }
+}
+
+
+public protocol NotificationEnum {
+    var name: Notification.Name { get }
+}
+
+
+extension NotificationCenter {
+    private static let _$key$_ = "_$key$_"
```

Comment:
Why the weird name?

Reply:
No good reason now. This type-safe use of notifications comes from something I worked out (https://github.com/wltrup/Swift-Typed-Notifications) after one of the objc.io talks and, at the time, for reasons I can't fathom now, I used that weird name. Given that it's private, I should - and will - change it to something more sensible.

Reply:
Fixed in https://github.com/Babylonpartners/DrawerKit/pull/36/commits/dedc463d71bee9babf44a73759bebe4f063b4239.